test_integration/03_redeem_and_unstake/utils/redeem_assertion.js

```diff
@@ -0,0 +1,209 @@
+ *                       transfers.
+ * @property {BN} nonce Redeem nonce.
+ * @property {string} beneficiary Address of beneficiary on origin chain.
+ * @property {string} hashLock Hash Lock provided by the redeemer.
+ * @property {string} unlockSecret Unlock secret to unlock hash lock.
+ * @property {string} messageHash Identifier for redeem and unstake process.
+ * @property {BN} blockHeight Height at which anchor state root is done.
+ */
+
+/**
+ * BaseToken(ETH) and OSTPrime ERC20 balance of cogateway, redeemer.
+ * @typedef {Object} Balances
+ * @property balances.ostPrime.cogateway ERC20 balance of cogateway contract.
+ * @property balances.ostPrime.redeemer ERC20 balance of beneficiary.
+ * @property balances.baseToken.cogateway Base token(ETH) balance of cogateway.
+ * @property balances.baseToken.redeemer Base token(ETH) balance of redeemer.
+ */
+
+/**
+ * Class to assert event and balances after Redeem.
+ */
+class RedeemAssertion {
+    /**
+     * Constructor.
+     * @param {Object} cogateway Truffle cogateway instance.
+     * @param {Object} ostPrime Truffle token instance.
+     * @param {Web3} web3 Web3 instance.
+     */
+    constructor(cogateway, ostPrime, web3) {
+        this.cogateway = cogateway;
+        this.token = ostPrime;
+        this.web3 = web3;
+    }
+
+    /**
+     * This verifies event and balances.
+     * @param {Object} event Event object after decoding.
+     * @param {Object} event Event object after decoding.
+     * @param {number} transactionFees Transaction fees in redeem request.
+     * @param {Balances} initialBalances Initial baseToken and token balances.
+     */
+    async verify(event, redeemRequest, transactionFees, initialBalances) {
+        await this._assertBalancesForRedeem(
+            redeemRequest,
+            initialBalances,
+            transactionFees,
+        );
+
+        RedeemAssertion._assertRedeemEvent(event, redeemRequest);
+    }
+
+    /**
+     * This captures base token and token balance of cogateway and redeemer
+     * @param {string} redeemer Redeemer address.
+     * @return {Promise<Balances>}
+     */
+    async captureBalances(redeemer) {
+        return {
+            baseToken: {
+                cogateway: await this._getEthBalance(this.cogateway.address),
+                redeemer: await this._getEthBalance(redeemer),
+            },
+            token: {
+                cogateway: await this.token.balanceOf(this.cogateway.address),
+                redeemer: await this.token.balanceOf(redeemer),
+            },
+        };
+    }
+
+    /**
+     * This asserts balances of redeemer and cogateway after Redeem.
+     * @param redeemRequest Redeem request parameters.
+     * @param {Balances} initialBalances Initial balance of redeemer and cogateway
+     *                                   generated by captureBalances method.
+     * @param {number} transactionFees Transaction fees in redeem request.
```

Comment:
This should be number or BN ?
```suggestion
     * @param {BN} transactionFees Transaction fees in redeem request.
```